packages/zowe-explorer/src/dataset/ZoweDatasetNode.ts

```diff
@@ -449,6 +450,17 @@ export class ZoweDatasetNode extends ZoweTreeNode implements IZoweDatasetTreeNod
         return this.etag;
     }
 
+    /**
+     * Returns the imperative.IProfileLoaded profile for this node
+     *
+     * @returns {imperative.IProfileLoaded}
+     */
+    public getProfile(): imperative.IProfileLoaded {
+        const prof = this.profile ?? this.getParent()?.getProfile();
```

Comment:
Should we have a little trace-log here?
`ZoweLogger.trace("ZoweDatasetNode.getProfile called.");`